discovery: Log KMS access denied errors as warning

juliaogris opened the pull request:
Log KMS access denied errors as warning, rather than returning an error,
as this features too prominently in the web UI and it is quite common
that KMS keys have strict policy for accessing tags or key policies even
if they can be listed.

camscale commented on lib/srv/discovery/fetchers/aws-sync/kms.go:
```diff
@@ -133,33 +136,57 @@ func (a *Fetcher) fetchKMSKeysForRegion(ctx context.Context, client kmsClient, r
+
+func isAccessDeniedException(err error) bool {
+	var apiErr smithy.APIError
+	if errors.As(err, &apiErr) {
+		return apiErr.ErrorCode() == "AccessDeniedException"
+	}
+	return false
```
nit: perhaps just:
```
return errors.As(err, &apiErr) && apiErr.ErrorCode() == "AccessDeniedException"
```

up to you if you prefer the longer form

camscale commented on lib/srv/discovery/fetchers/aws-sync/aws-sync.go:
```diff
@@ -64,6 +65,8 @@ type Config struct {
 	// DiscoveryConfigName if set, will be used to report the Discovery Config Status to the Auth Server.
 	DiscoveryConfigName string
+	// Log is the logger to use for logging.
+	Log *slog.Logger
```
Since this logger is required, I think you need to add a check to `CheckAndSetDefaults()`

juliaogris committed:
discovery: Log KMS access denied errors as warning

Log KMS access denied errors as warning, rather than returning an error,
as this features too prominently in the web UI and it is quite common
that KMS keys have strict policy for accessing tags or key policies even
if they can be listed.
